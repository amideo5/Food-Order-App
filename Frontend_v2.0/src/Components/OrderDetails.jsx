import React, { useEffect, useState } from "react";
import axios from "axios";
import NavigationBar from "./NavigationBar";
import "../assets/welcome.css";
import NavBar from "./NavBar";
import { Card, Button } from "react-bootstrap";

const OrderDetails = () => {
  const [orders, setOrders] = useState({
    items: {
      orderItems: {},
    },
  });
  const email = localStorage.getItem("email");

  useEffect(() => {
    getName();
  }, []);

  async function getName() {
    let name = await axios.get("http://localhost:8080/users/name/" + email);

    console.log(name);

    let data = await axios.get(
      "http://localhost:8080/orders/getOrder/" + name.data
    );

    console.log(data.data.items.orderItems);
    Object.keys(data.data.items.orderItems).map((key) => {
      console.log(data.data.items.orderItems[key]);
    });
    setOrders(data.data);
  }

  if (localStorage.getItem("token") != null) {
    return (
      <>
        <div className="welcomePage">
          <NavigationBar />

          <Card className={"border border-dark bg-light text-dark paddingTop"}>
            <Card.Header>
              <h1>Order Summary</h1>
            </Card.Header>
          </Card>

          <br />
          <br />
          <br />

          <Card style={{ width: "18rem" }}>
            <Card.Body>
              <Card.Title>{orders.orderId}</Card.Title>
              <br />
              <strong>Order Name :</strong> {orders.orderName}
              <br />
              <strong>Order Items :</strong>
              {Object.keys(orders.items.orderItems).map((key) => {
                return (
                  <div>
                    {key} : {orders.items.orderItems[key]}
                  </div>
                );
              })}
              <strong>Order Cost :</strong> {orders.cost}
            </Card.Body>
          </Card>
        </div>
      </>
    );
  }
  if (localStorage.getItem("token") == null) {
    return (
      <>
        <NavBar />
        <div className="form-container">
          You have to login to use this application
        </div>
      </>
    );
  }
};

export default OrderDetails;
