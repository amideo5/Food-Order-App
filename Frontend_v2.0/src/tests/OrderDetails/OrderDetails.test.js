import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";
import OrderDetails from "../../Components/OrderDetails";
import { fireEvent, waitFor } from "@testing-library/react";

describe("Testing Order Details Component", () => {
  test("Snapshot testing", () => {
    const { container } = render(<OrderDetails />);

    expect(container).toMatchSnapshot();
  });

  test("Checking component is rendering or not", () => {
    render(<OrderDetails />);
    localStorage.setItem("token", "sdjs");
    const navText = screen.getAllByText("NEX Food Ordering");
    expect(navText[0]).toBeInTheDocument();
  });

  test("Normal flow", async () => {
    localStorage.setItem("email", "adn@gmail.com");
    render(<OrderDetails />);

    act(async ()=>{
        expect( await screen.findByText("Aryan")).toBeInTheDocument()
        expect(await screen.findByText("rajma")).toBeInTheDocument();
    })

  });
});
