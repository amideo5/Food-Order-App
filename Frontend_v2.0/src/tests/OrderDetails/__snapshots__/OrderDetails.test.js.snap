// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Order Details Component Snapshot testing 1`] = `
<div>
  <nav
    class="m-0 py-3 navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        NEX Food Ordering
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        />
        <a
          class="nav-link"
          href="/admin"
        >
          ADMIN
        </a>
        <a
          class="nav-link"
          href="/"
        >
          Sign In
        </a>
        <a
          class="nav-link"
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
  <div
    class="form-container"
  >
    You have to login to use this application
  </div>
</div>
`;
